### PR TITLE
pmd:ConsecutiveAppendsShouldReuse - Consecutive Appends Should Reuse

### PR DESCRIPTION
**Pull request by m-ezzat**
This pull request is focused on resolving occurrences of Sonar rule pmd:ConsecutiveAppendsShouldReuse - Consecutive Appends Should Reuse
You can find more information about the issue here:
https://dev.eclipse.org/sonar/coding_rules#q=pmd:ConsecutiveAppendsShouldReuse
Please let me know if you have any questions.
M-Ezzat

### CODE REVIEW

File: jodd-core/src/main/java/jodd/datetime/DateTimeStamp.java [hunk 1]
```diff
@@ -228,8 +228,9 @@ public int compareDateTo(Object o) {
 	@Override
 	public String toString() {
 		StringBuilder sb = new StringBuilder(25);
-		sb.append(year).append('-').append(month).append('-').append(day).append(' ');
-		sb.append(hour).append(':').append(minute).append(':').append(second).append('.').append(millisecond);
+		sb.append(year).append('-').append(month).append('-').append(day)
+				.append(' ').append(hour).append(':').append(minute)
+				.append(':').append(second).append('.').append(millisecond);
```
Comment (slandelle) [inline jodd-core/src/main/java/jodd/datetime/DateTimeStamp.java]: return new StringBuilder().append()....toString();

File: jodd-joy/src/main/java/jodd/joy/page/PageData.java [hunk 2]
```diff
@@ -205,13 +205,10 @@ public boolean isFirstPage() {
...
-		result.append(currentPage).append(',');
-		result.append(totalPages).append(',');
-		result.append(totalItems).append(',');
-		result.append(firstIndex).append(',');
-		result.append(lastIndex).append(',');
-		result.append(pageSize).append(']');
+		result.append('[').append(currentPage).append(',').append(totalPages)
+				.append(',').append(totalItems).append(',').append(firstIndex)
+				.append(',').append(lastIndex).append(',').append(pageSize)
+				.append(']');
```
Comment (slandelle) [inline jodd-joy/src/main/java/jodd/joy/page/PageData.java]: return new StringBuilder().append()....toString();

File: jodd-jtx/src/main/java/jodd/jtx/JtxTransactionMode.java [hunk 3]
```diff
@@ -249,11 +249,10 @@ public int hashCode() {
...
 		StringBuilder sb = new StringBuilder();
-		sb.append("jtx{");
-		sb.append(propagationBehavior.toString());
-		sb.append(',').append(readOnlyMode ? "readonly" : "readwrite");
-		sb.append(',').append(isolationLevel.toString());
-		sb.append(',').append(timeout).append('}');
+		sb.append("jtx{").append(propagationBehavior.toString()).append(',')
+				.append(readOnlyMode ? "readonly" : "readwrite").append(',')
+				.append(isolationLevel.toString()).append(',').append(timeout)
+				.append('}');
```
Comment (slandelle) [inline jodd-jtx/src/main/java/jodd/jtx/JtxTransactionMode.java]: return new StringBuilder().append()....toString();

File: jodd-log/src/main/java/jodd/log/impl/SimpleLogger.java [hunk 4]
```diff
@@ -109,17 +109,10 @@ protected void print(Level level, String message, Throwable throwable) {
...
-		msg.append(' ');
-		msg.append(slf.getCallerClass());
-		msg.append(' ');
-		msg.append('-');
-		msg.append(' ');
-		msg.append(message);
+		msg.append(slf.getElapsedTime()).append(' ').append('[')
+				.append(level.toString()).append(']').append(' ')
+				.append(slf.getCallerClass()).append(' ').append('-')
+				.append(' ').append(message);
```
Comment (slandelle) [inline jodd-log/src/main/java/jodd/log/impl/SimpleLogger.java]: return new StringBuilder().append()....toString();

File: jodd-swingspy/src/main/java/jodd/swingspy/SwingSpyPanel.java [hunk 5]
```diff
@@ -179,8 +179,10 @@ public String toString() {
 
 		public String toHtmlString() {
 			StringBuilder str = new StringBuilder("<html>");
-			str.append("&nbsp;name: ").append("<b>").append(component.getName()).append("</b><br>");
-			str.append("&nbsp;class: ").append("<b>").append(component.getClass().getName()).append("</b><br>");
+			str.append("&nbsp;name: ").append("<b>")
+					.append(component.getName()).append("</b><br>")
+					.append("&nbsp;class: ").append("<b>")
+					.append(component.getClass().getName()).append("</b><br>");
```
Comment (slandelle) [inline jodd-swingspy/src/main/java/jodd/swingspy/SwingSpyPanel.java]: return new StringBuilder().append()....toString();

**Comment by igorspasic** [comment]
got it! Thanx @slandelle !

File: jodd-core/src/main/java/jodd/util/BCrypt.java [hunk 6]
```diff
@@ -736,11 +736,11 @@ public static String hashpw(String password, String salt) {
 			throw new IllegalArgumentException(
 				"rounds exceeds maximum (30)");
 		}
-		rs.append(Integer.toString(rounds));
-		rs.append("$");
-		rs.append(encode_base64(saltb, saltb.length));
-		rs.append(encode_base64(hashed,
-			bf_crypt_ciphertext.length * 4 - 1));
+		rs.append(Integer.toString(rounds))
```
Comment (slandelle) [inline jodd-core/src/main/java/jodd/util/BCrypt.java]: bad: StringBuilder.append(int) exists and is optimized

File: jodd-core/src/main/java/jodd/util/BCrypt.java [hunk 7]
```diff
@@ -767,9 +767,8 @@ public static String gensalt(int log_rounds, SecureRandom random) {
 			throw new IllegalArgumentException(
 				"log_rounds exceeds maximum (30)");
 		}
-		rs.append(Integer.toString(log_rounds));
-		rs.append("$");
-		rs.append(encode_base64(rnd, rnd.length));
+		rs.append(Integer.toString(log_rounds)).append("$")
```
Comment (slandelle) [inline jodd-core/src/main/java/jodd/util/BCrypt.java]: bad: StringBuilder.append(int) exists and is optimized

File: jodd-jtx/src/main/java/jodd/jtx/JtxTransactionMode.java [hunk 8]
```diff
@@ -249,11 +249,10 @@ public int hashCode() {
 	@Override
 	public String toString() {
 		StringBuilder sb = new StringBuilder();
-		sb.append("jtx{");
-		sb.append(propagationBehavior.toString());
-		sb.append(',').append(readOnlyMode ? "readonly" : "readwrite");
-		sb.append(',').append(isolationLevel.toString());
-		sb.append(',').append(timeout).append('}');
+		sb.append("jtx{").append(propagationBehavior.toString()).append(',')
```
Comment (slandelle) [inline jodd-jtx/src/main/java/jodd/jtx/JtxTransactionMode.java]: Useless `toString()`, that's what StringBuilder.append(Object) does (just that i's protected against NPE and will print "null").

File: jodd-log/src/main/java/jodd/log/impl/SimpleLogger.java [hunk 9]
```diff
@@ -109,17 +109,10 @@ protected void print(Level level, String message, Throwable throwable) {
...
-		msg.append(level.toString());
-		msg.append(']');
-		msg.append(' ');
-		msg.append(slf.getCallerClass());
-		msg.append(' ');
-		msg.append('-');
-		msg.append(' ');
-		msg.append(message);
+		msg.append(slf.getElapsedTime()).append(' ').append('[')
+				.append(level.toString()).append(']').append(' ')
```
Comment (slandelle) [inline jodd-log/src/main/java/jodd/log/impl/SimpleLogger.java]: Useless toString()

**Comment by igorspasic** [comment]
Just started reviewing :)<br>:rocket:

**Comment by igorspasic** [comment]
Very nice @m-ezzat! Especially loved the part in `JDateTime` where you calculated the `StringBuilder` size! Kudos!!!

**Pull request merged**